lib/common/pool.c

```diff
@@ -225,13 +225,16 @@ static POOL_ctx* POOL_resize_internal(POOL_ctx* ctx, size_t numThreads)
  *    note : starting context is considered consumed. */
 POOL_ctx* POOL_resize(POOL_ctx* ctx, size_t numThreads)
 {
-    POOL_ctx* newCtx;
     if (ctx==NULL) return NULL;
     ZSTD_pthread_mutex_lock(&ctx->queueMutex);
-    newCtx = POOL_resize_internal(ctx, numThreads);
+    {   POOL_ctx* const newCtx = POOL_resize_internal(ctx, numThreads);
+        if (newCtx!=ctx) {
+            POOL_free(ctx);
```

Comment:
Destroying a locked mutex is undefined behavior. http://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html

Reply:
I'll change the behavior then, and just use an error code return type.
`free` will be an external operation.